notes/lecture21.tex

```diff
@@ -0,0 +1,192 @@
+\begin{definition}
+The matrix $A$ satisfies the restricted nullspace property (RNP) with respect to the support $S$ if $C(S) \cup \text{null}(A) = \{ 0 \}$.
+\end{definition}
+With these definitions in place, we can now state our main theorem.
+\begin{theorem}
+Given $A \in \R^{n \times d}$ and $y \in \R^n$ we consider the solution to the $l_0$-minimization problem $x^* = \argmin_{Ax=y}\|x\|_0$. Assume $x^*$ has support $S$ and let the matrix $A$ satisfy the restricted nullspace property with respect to S. Then given the solutions of the $l_1$-minimization problem $\hat{x} = \argmin_{Ax=y}\|x\|_1$ we have $\hat{x} = x^*$.
+\end{theorem}
+
+\begin{proof}
+We first note that by definition both $x^*$ and $\hat{x}$ satisfy our feasibility constraint $Ax=y$. Letting $\Delta = \hat{x} - x^*$ be the error vector we have $A\Delta = A\hat{x} - Ax^* = 0$, which implies that $\Delta \in null(A)$.
+
+Our goal now is to show that $\Delta \in C(S)$ then we would have $\Delta = 0$ from the restricted nullspace property. First, since $\hat{x}$ is optimal in $l_1$ it follows that $\|\hat{x}\|_1 \le \|x^*\|_1$. We then have
+\begin{align*}
+    \|x_S^*\|_1 &= \|x^*\|_1 \ge \|\hat{x}\|_1\\
+                &= \|x^* + \Delta\|_1\\
+                &= \|x_S^* + \Delta_S\|_1 + \|x_{\overline{S}}^* + \Delta_{\overline{S}}\|_1 &&\text{by splitting the $l_1$ norm,}\\                            
+                &= \|x_S^* + \Delta_S\|_1 + \|\Delta_{\overline{S}}\|_1 &&\text{by the support assumption of $\|x^*\|_1$,}\\
+                &\ge \|x_S^*\|_1 - \|\Delta_S\|_1 + \|\Delta_{\overline{S}}\|_1.
+\end{align*}
+Hence $\|\Delta_S\|_1 \ge \|\Delta_{\overline{S}}\|_1$, which implies $\Delta \in C(S)$.
+\end{proof}
+
+So far, so good. We have shown that the $l_1$-relaxation works for certain matrices. A natural question however is what kinds of matrices satisfy the restricted nullspace property. In order to get a handle on this, we will study yet another nice property of matrices, the so called restricted isometry property (RIP). Later, we will then see that specific matrix ensembles satisfy RIP with high probability.
+\begin{definition}
+A matrix $A$ satisfies the $(s, \delta)$-RIP if for all $s$-sparse vectors $x$ ($\|x\|_0 \le s$), we have
+\[
+(1 - \delta)\|x\|_2^2 \le \|Ax\|_2^2 \le (1 + \delta) \|x\|_2^2.
+\]
+\end{definition}
+The intuition is that $A$ acts like an isometry on sparse vectors (a true isometry would have $\delta = 0$). The RIP is useful since it implies that the difference between two $s$-sparse vectors cannot be mapped to $0$. By looking at the singular values of $A$ we can derive the following lemma.
+\begin{lemma}
+\label{RIPlemma}
+If $A$ has $(s, \delta)$-RIP, then
+\[
+\|A_S^{\trans}A_S - I\|_2 \le \delta
+\]
+for all subsets $S$ of size $s$. Where
+
+\begin{align*}
+    (A_S)_{ij} = \begin{cases}
+    A_{ij} & \text{if } i \in S \\
```

Comment:
$i \in S$ should be $j \in S$

Reply:
👍

Reply:
Here's an updated pull request with your changes.